src/commands/qrcode.cpp

```diff
@@ -23,25 +23,172 @@
 #include <iostream>
 #include <bitcoin/explorer/define.hpp>
 
+#if defined WITH_QRENCODE && defined WITH_LIBPNG
+#include "png.h"
+#include "qrencode.h"
+#endif
+
 using namespace bc::explorer;
 using namespace bc::explorer::commands;
 
+/*
+  The following constants and write_png method are based on:
+
+  https://github.com/fukuchi/libqrencode/blob/3.4/qrenc.c
+*/
+
+#if defined WITH_QRENCODE && defined WITH_LIBPNG
+static constexpr int dpi = 72;
+static constexpr int version = 0;
+static constexpr int margin = 2;
+static constexpr QRencodeMode hint = QR_MODE_8;
+static constexpr QRecLevel level = QR_ECLEVEL_L;
+static constexpr int casesensitive = 1;
+static constexpr int INCHES_PER_METER = (100.0 / 2.54);
+static constexpr unsigned int fg_color[4] = {0, 0, 0, 255};
+static constexpr unsigned int bg_color[4] = {255, 255, 255, 255};
+
+
+static bool write_png(QRcode *qrcode, uint32_t size, std::string outfile);
+#endif
+
 console_result qrcode::invoke(std::ostream& output, std::ostream& error)
 {
+#if defined WITH_QRENCODE && defined WITH_LIBPNG
     // Bound parameters.
-    //const auto& address = get_payment_address_argument();
+    const auto& address = get_payment_address_argument();
+    const auto& filename = get_filename_argument();
+    const auto& size = get_size_option();
 
-    error << BX_QRCODE_NOT_IMPLEMENTED << std::endl;
+    std::string qr_string = "bitcoin:" + address.encoded();
+
+    QRcode* qrcode = QRcode_encodeString(
+        qr_string.c_str(), version, level, hint, casesensitive);
```

Comment:
This should be implemented in `libbitcoin::wallet` (e.g `qrcode.h/cpp`). We may eventually add a decoder to this as well. The wrapper signature should be in the libbitcoin style (i.e. using explicitly-sized numeric types, class enumerations, typdef'd color, data_chunk& return value, etc.)